pkg/grpc/server.go

```diff
@@ -0,0 +1,115 @@
+package grpc
+
+import (
+	//"context"
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+
+	//"go.uber.org/zap/zapcore"
+
+	//igrpc "get.porter.sh/porter/gen/proto/go/porterapis/installation/v1alpha1"
+	pGRPC "get.porter.sh/porter/gen/proto/go/porterapis/porter/v1alpha1"
+	"get.porter.sh/porter/pkg/config"
+	pserver "get.porter.sh/porter/pkg/grpc/portergrpc"
+	"get.porter.sh/porter/pkg/porter"
+	"get.porter.sh/porter/pkg/tracing"
+
+	//"go.opentelemetry.io/otel/attribute"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/reflection"
+)
+
+var (
+	reg = prometheus.NewRegistry()
+	// Create some standard server metrics.
+	grpcMetrics = grpc_prometheus.NewServerMetrics()
+
+	// Create a customized counter metric.
+	customizedCounterMetric = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "demo_server_say_hello_method_handle_count",
+		Help: "Total number of RPCs handled on the server.",
+	}, []string{"name"})
+)
+
+func init() {
+	reg.MustRegister(grpcMetrics, customizedCounterMetric)
+	customizedCounterMetric.WithLabelValues("Test")
+}
+
+type PorterGRPCService struct {
+	PorterConfig *config.Config
+	opts         *porter.ServiceOptions
+}
+
+/*
+type Config struct {
+	Port        int    `mapstructure:"grpc-port"`
+	ServiceName string `mapstructure:"grpc-service-name"`
+}
+*/
+
+func NewServer(ctx context.Context, opts *porter.ServiceOptions) (*PorterGRPCService, error) {
+	pCfg := config.New()
+	srv := &PorterGRPCService{
+		PorterConfig: pCfg,
+		opts:         opts,
+	}
+	return srv, nil
+}
+
+func (s *PorterGRPCService) ListenAndServe() (*grpc.Server, error) {
+	ctx := context.TODO()
+	_, log := tracing.StartSpan(ctx)
```

Comment:
although `context.TODO()` and `context.Background()` are very similar, I would either do use the latter here, or better yet add a `Context` to the `PorterGRPCService` and build a child of it here. for example, assuming there were a `ctx` field on `s`:

```suggestion
	_, log := tracing.StartSpan(s.ctx)
```

Reply:
I'd like to still get the context into PorterGRPCService